packages/opentelemetry-tracing/test/export/TestTracingSpanExporter.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  BasicTracerProvider,
+  SpanExporter,
+  ReadableSpan,
+  Tracer,
+  SpanProcessor,
+} from '../../src';
+import { ExportResult, NoopLogger, AlwaysOnSampler } from '@opentelemetry/core';
+
+/**
+ * A test-only span exporter that naively simulates triggering instrumentation
+ * (creating new spans) during export.
+ */
+export class TestTracingSpanExporter implements SpanExporter {
```

Comment:
We already have an in memory span exporter for this type of testing. Why did you not use that?

Reply:
I could-have/can extend it. Ultimately, have to add extra logic (unless change approach) to do instrumentation to simulate making web calls, etc. As I was fleshing this out, wasn't sure how much would deviate but didn't end up being much different.

Reply:
Ah the spans, I forgot. Maybe just `extend` it and call `super.export()` in your export method.

Reply:
yeah, makes sense. sort of lost track of how simple it molded back down to.